src/app/static/src/app/types.ts

```diff
@@ -154,4 +154,43 @@ export interface UploadFile {
     resourceId: string | null,
     resourceUrl: string | null,
     lastModified: string | null
-}
\ No newline at end of file
+}
+
+export interface DatasetConfig {
+    id: string,
+    type: "standard" | "custom",
+    label: string,
+    module?: string,
+    prop?: string,
+    filters?: Filter[]
+}
+
+export interface DataSourceConfig {
+    id: string,
+    type: "fixed" | "editable",
+    label: string,
+    datasetId: string
+    showFilters: true
+    showIndicators: true
+}
+
+export interface GenericChartMetadata {
+    datasets: DatasetConfig[],
+    dataSelectors: {
+        dataSources: DataSourceConfig[]
+    },
+    subplots?: {
+        columns: number,
+        distinctColumn: string,
+        heightPerRow: number
+    },
+    chartConfig: {
+        id: string,
+        label: string,
+        config: string
+    }[]
+}
+
+export interface GenericChartMetadataResponse {
+    [key: string]: GenericChartMetadata;
```

Comment:
We seem to be vacillating between `,`, `;` and nothing for EOL separators in interfaces. But it makes no difference, as far as I know!

Reply:
OK, let's try to consistify with the simplest case, no line endings.